src/validation.cpp

```diff
@@ -5009,15 +5009,6 @@ void ChainstateManager::Unload()
     m_best_invalid = nullptr;
 }
 
-void ChainstateManager::Reset()
```

Comment:
I make use of this in #24232

Reply:
Ok, reverted to the original version in 686d35de6 that keeps Reset() and replaces the lock with annotation+assertion.

Reply:
(not saying what this pull should do, just saying what the long term goal should be)

Reply:
This was not just "test-only code" - the Reset() was used to detect snapshot dirs that needed to be cleaned up. The merge here isn't now just a matter of copy/pasting some code.